src/main/scala/com/fulcrumgenomics/fastq/FastqToBam.scala

```diff
@@ -149,6 +150,7 @@ class FastqToBam
       val subs = fqs.iterator.zip(structures.iterator).flatMap { case(fq, rs) => rs.extract(fq.bases, fq.quals) }.toIndexedSeq
       val sampleBarcode = subs.iterator.filter(_.kind == SampleBarcode).map(_.bases).mkString("-")
       val umi           = subs.iterator.filter(_.kind == MolecularBarcode).map(_.bases).mkString("-")
+      val umiQual       = subs.iterator.filter(_.kind == MolecularBarcode).map(_.quals).mkString(" ")
```

Comment:
Any reason not to use the same separator as the bases?
```suggestion
      val umiQual       = subs.iterator.filter(_.kind == MolecularBarcode).map(_.quals).mkString("-")
```

Reply:
Yes, there's a reason: because the [official specification](https://samtools.github.io/hts-specs/SAMtags.pdf) says it.
The tag that should contain the qualities of the bases in the _RX_ tag is the _QX_ tag. And the description of that tag (page 6) literally says:
> In the case of multiple unique molecular identifiers (e.g., one on each end of the template) the recommended implementation concatenates all the quality strings with a space (‘ ’) between the different strings.

It is just a recommendation, but unless we offer the user a way to customize the separator, I would go for the recommended one (like we are following the specs by using an hyphen as a separator in the _RX_ tag).

Reply:
Got it.  Fixed `AnnotateBamWithUmis` to have the same behavior, as well as fixed a bug: https://github.com/fulcrumgenomics/fgbio/pull/742.

Reply:
I didn't remember having tweaked this behavior in `AnnotateBamWithUmis`, I just assumed the code was OK and added a very similar code for the qualities.
Either case, now that the issue is identified, it should be easy to solve :-)